sdk/core/azure-core/src/http/winhttp/win_http_request.hpp

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+// cspell:words PCCERT HCERTSTORE
+
+/**
+ * @file
+ * @brief #Azure::Core::Http::HttpTransport request support classes.
+ */
+
+#pragma once
+
+#include "azure/core/http/win_http_transport.hpp"
+#include "azure/core/url.hpp"
+#include <Windows.h>
+#include <memory>
+#include <mutex>
+#include <wil\resource.h>
+#include <winhttp.h>
+
+namespace Azure { namespace Core { namespace Http { namespace _detail {
+
+  class WinHttpRequest;
+  /**
+   * @brief An outstanding WinHTTP action. This object is used to process asynchronous WinHTTP
+   * actions.
+   *
+   * The WinHttpRequest object has a WinHttpAction associated with it to convert asynchronous
+   * WinHTTP operations to synchronous operations.
+   *
+   */
+  class WinHttpAction final {
+    // The WinHttpRequest class is friend because it needs to have access to StatusCallback when it
+    // registers for notifications, but there's no other reason to make it public.
```

Comment:
In that case, given this whole header is private, I would consider marking `StatusCallback()` public, so that `WinHttpRequest` only has access to that targeted method, and doesn't have access to all of the private methods and members of `WinHttpAction`. That seems like a much smaller access scope between the two types.